File: instedit/reset_instrument_preview.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

/*static*/ void reset_instrument_preview (void)
{
  uint16_t ins;

  if (play_status != isPlaying) reset_player ();
  ins = check_4op_to_test ();
  if (ins == 0)
    update_instr_data (instrum_page);
  else
  {
    update_instr_data (ins & 0xFF);
    update_instr_data (ins >> 8);
  }
}

File: instedit/pas/reset_instrument_preview.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure reset_instrument_preview;

var
  temp: Word;

begin
  If (play_status <> isPlaying) then reset_player;
  temp := check_4op_to_test;
  If NOT (temp <> 0) then
    update_instr_data(instrum_page)
  else begin
         update_instr_data(LO(temp));
         update_instr_data(HI(temp));
       end;
end;
